src/main/java/com/miraisolutions/xlconnect/Workbook.java

```diff
@@ -1566,10 +1612,10 @@ public void clearRangeFromReference(String reference) {
         clearRange(sheetName, coords);
     }
     
-    public void clearNamedRegion(String name) {
-        String sheetName = getName(name).getSheetName();
-        int[] coords = getReferenceCoordinates(name);
-        clearRange(sheetName, coords);
+    public void clearNamedRegion(String name, String worksheetName) {
+        String dataSourceSheetName = getName(name).getSheetName();
+        int[] coords = getReferenceCoordinatesForName(worksheetName, name);
+        clearRange(dataSourceSheetName, coords);
```

Comment:
Argument name, used also to get the `Name`.
```suggestion
    public void clearNamedRegion(String name, String worksheetScope) {
        String dataSourceSheetName = getNameInWorksheetScope(worksheetScope, name).getSheetName();
        int[] coords = getReferenceCoordinatesForName(worksheetScope, name);
        clearRange(dataSourceSheetName, coords);
```

Reply:
thanks, I missed that one